staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -60,10 +84,14 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 			req = req.WithContext(authenticator.WithAudiences(req.Context(), apiAuds))
 		}
 		resp, ok, err := auth.AuthenticateRequest(req)
-		if err != nil || !ok {
-			if err != nil {
-				klog.Errorf("Unable to authenticate the request due to an error: %v", err)
-			}
+		if err != nil {
+			klog.Errorf("Unable to authenticate the request due to an error: %v", err)
+			authenticationErrorCounter.Inc()
+			failed.ServeHTTP(w, req)
+			return
+		}
+		if !ok {
+			authenticationFailureCounter.Inc()
 			failed.ServeHTTP(w, req)
```

Comment:
It is probably safer to avoid multiple paths that return:

```go
resp, ok, err := auth.AuthenticateRequest(req)
if err != nil || !ok {
	if err != nil {
		klog.Errorf("Unable to authenticate the request due to an error: %v", err)
		authenticationErrorCounter.Inc()
	}
	if !ok {
		authenticationFailureCounter.Inc()
	}

	failed.ServeHTTP(w, req)
	return
}
```

Reply:
Done.